### ISSUE TITLE
Tree is not search engine friendly

### ISSUE THREAD

**Issue by stuporglue** [dovy/Pedigree-Viewer#2]
Pedigree should be printed out as static content in a <noscript> block for search engines and screen readers. It's probably even Ok if it's not in a tree format.

**Comment by dovy** [comment]
That's fine if it's one GEDCOM per page load, but what about dynamic content? Now you're going to basically output the entire gedcom to the HTML dom before making the JS tree. Is that within the scope of a pedigree viewer?

**Comment by stuporglue** [comment]
It would be a single GEDCOM, and it probably needs to be the whole gedcom too -- at least names, dates and places of individuals. Each person wouldn't need to be in it's own div, or even it's own <p> necessarily. If they were all in a single div, the load on the DOM wouldn't be that great. 

So, the DOM would contain this text summary of the entire GEDCOM file in <noscript> tags and the Pedigree-Viewer tree would contain a subset (20-100ish individuals). This would keep the load on the DOM low, even in extensive trees while giving visual users and non-visual uses the info they need. 

Users aren't going to see this content (unless they have JavaScript disabled) it's mainly for Search Engines with a secondary goal of making it somewhat accessible to screen readers. 

I care about search engines because when I put up my family history website, it's with a goal of sharing my info and making it findable to others.

**Comment by dovy** [comment]
Hmm, you're thinking very GEDCOM based. What about those who use a database? Or the results from an API? 

Just saying if you have like a GEDCOM file of some 3000 names, that's going to take a while to generate a file to utilize. Also then using json to parse that file and get the proper nodes created will be a memory beast. This may be a side step into your PHP gedcom parser. I'd suggest a cache of json and a cache of html for including in the page. Not sure the best approach here.

**Comment by stuporglue** [comment]
My goal is that this will be something that non-programmers who are minimally technically savvy can use. Ideally if you're technical enough to find this project via Google, then you would be technical enough to use it. So, the non-programmer genealogist who has a website should be able to unzip the project, plop in a GEDCOM and have it "just work". 

The more technical can modify makeJson.php to connect to an API, a database, or whatever they want. Or they can modify the call to drawChart() in index.php to call a static file, a ruby, python or .NET script. 

GEDCOM has its flaws, but it's the lowest common denominator. 

## 

As an aside, the PHP gedcom parser is pretty fast. If you can get me a big gedcom file, I bet the times would be acceptable, even in a 3000 name file.

**Comment by stuporglue** [comment]
"Is that within the scope of a pedigree viewer?" 

Upon further review, probably not. Print the info you want on your own page. Just use this for tree display
